spring-statemachine-uml/src/main/java/org/springframework/statemachine/plantuml/helper/NameGetter.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2002-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.statemachine.plantuml.helper;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.apache.commons.lang3.ClassUtils;
+import org.apache.commons.lang3.RegExUtils;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.BeanNameAware;
+import org.springframework.expression.Expression;
+import org.springframework.lang.Nullable;
+import org.springframework.statemachine.action.Action;
+import org.springframework.statemachine.action.SpelExpressionAction;
+import org.springframework.statemachine.guard.Guard;
+import org.springframework.statemachine.guard.SpelExpressionGuard;
+
+import java.lang.reflect.Field;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class NameGetter {
+
+	private static final Log log = LogFactory.getLog(NameGetter.class);
+
+	/**
+	 * Implement a 'name strategy' based on this sequence:
+	 * <UL>
+	 * <LI>{@link Expression}</LI>
+	 * <LI>{@link BeanNameAware}</LI>
+	 * <LI>Lambda's unique "arg$1" parameter</LI>
+	 * </UL>
+	 * If all these strategy are failing, fallback to "class name" of the object
+	 *
+	 * @param object object to get "name" from
+	 * @return name of the object
+	 */
+	public static String getName(Object object) {
```

Comment:
This class should probably be compared to https://github.com/spring-projects/spring-statemachine/blob/26ef2bcdc99ad6351bffe6fd70fc46b54dddd5fa/spring-statemachine-autoconfigure/src/main/java/org/springframework/statemachine/boot/support/BootStateMachineMonitor.java#L137-L143
which seems to be the current mean to get "names" of Spring State Machine actions, guard, etc..